sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.createApplicationOfferingChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,417 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  authorizeUserTokenForLocation,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+import {
+  createFakeInstitutionLocation,
+  E2EDataSources,
+  createE2EDataSources,
+  saveFakeApplicationDisbursements,
+  createFakeEducationProgramOffering,
+  createFakeUser,
+} from "@sims/test-utils";
+import {
+  ApplicationOfferingChangeRequestStatus,
+  ApplicationStatus,
+  InstitutionLocation,
+  OfferingIntensity,
+} from "@sims/sims-db";
+import { createFakeSINValidation } from "@sims/test-utils/factories/sin-validation";
+import { addDays } from "@sims/utilities";
+
+describe("ApplicationOfferingChangeRequestInstitutionsController(e2e)-createApplicationOfferingChangeRequest", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let collegeFLocation: InstitutionLocation;
+  let institutionUserToken: string;
+  let endpoint: string;
+  const OLD_ENV = process.env;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    // Institution token.
+    institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    endpoint = `/institutions/location/${collegeFLocation.id}/application-offering-change-request`;
+
+    // To check the study overlap, the BYPASS_APPLICATION_SUBMIT_VALIDATIONS needs to be set false.
+    process.env = {
```

Comment:
Because there is no need to reset the env variables would not the below change be enough?
`process.env.BYPASS_APPLICATION_SUBMIT_VALIDATIONS = "false"`

Reply:
@andrewsignori-aot @ann-aot  the variable `BYPASS_APPLICATION_SUBMIT_VALIDATIONS` can be complete skipped right?

If it is not set, it is equivalent to false and also for gitHub actions we do not set this variable. 

I believe we have this variable set as true in local .env file and that is the reason we set to true while running locally.

Reply:
@andrewsignori-aot `process.env.BYPASS_APPLICATION_SUBMIT_VALIDATIONS = "false"` will also work. code updated.

@dheepak-aot  I am not sure about the variable in gitHub. But we need it local.

Reply:
By saying gitHub actions I meant tests workflow. We don't pass that in `repo-checks-tests.yml`. 

My point here is, should we add something here for the sake of running test in local instead of removing/commenting that property in .env file ?

Reply:
mm, @dheepak-aot , IMO, I don't see any issue to update an env variable for the sake of TC, I saw some examples in StackOverflow doing so

Reply:
@dheepak-aot I noticed the same and I see the point. I was ok to ensure that it was `false` to allow the tests to be executed locally (outside the docker environment) without the need to be changing the local variables.
What if we use the `.env-e2e-tests` to override the BYPASS_APPLICATION_SUBMIT_VALIDATIONS  to false? In this way running locally using make tests would still work without the need to change our `.env` file always, keeping in mind that we may need to do the same for other env variables also.
Having the change added to the  `.env-e2e-tests` will still demand the DEV to change the local .env file but only if they would be running the complete test suits locally, which I do not see the point if we have the docker enabled.